File: frontend/src/types/schedule.ts
export interface ScheduleSlot {
  id: number
  goal_id: number
  day: string
  start_time: string
  end_time: string
  note: string
  created_at: string
}

export interface CreateSlotRequest {
  goal_id?: number
  day: string
  start_time: string
  end_time: string
  note?: string
}

File: backend/internal/domain/schedule.go
package domain

import "time"

type ScheduleSlot struct {
	ID        int64     `json:"id"`
	GoalID    int64     `json:"goal_id"`
	Day       string    `json:"day"`        // monday, tuesday, ...
	StartTime string    `json:"start_time"` // "09:00"
	EndTime   string    `json:"end_time"`   // "10:00"
	Note      string    `json:"note"`
	CreatedAt time.Time `json:"created_at"`
}
